src/file-events/headers/generic_fsnotifier.h

```diff
@@ -23,7 +23,8 @@ enum FileWatchEventType {
     CREATED,
     REMOVED,
     MODIFIED,
-    INVALIDATE,
+    INVALIDATED,
+    OVERFLOWED,
```

Comment:
I suppose those types will go away when we move to the visitor kind of callback, right?

Reply:
Yes. That's the primary motivation for the visitor-style API.